LongWarOfTheChosen/Src/LW_PerkPack_Integrated/Classes/X2Effect_SteadyWeapon.uc

```diff
@@ -105,18 +110,49 @@ static function EventListenerReturn SteadyWeaponWoundListener(Object EventData,
 function GetToHitModifiers(XComGameState_Effect EffectState, XComGameState_Unit Attacker, XComGameState_Unit Target, XComGameState_Ability AbilityState, class<X2AbilityToHitCalc> ToHitType, bool bMelee, bool bFlanking, bool bIndirectFire, out array<ShotModifierInfo> ShotModifiers)
 {
 	local ShotModifierInfo ShotInfo;
+	local XComGameState_Item SourceWeaponState;
+
+	SourceWeaponState = AbilityState.GetSourceWeapon();
 
 	if(!bMelee && AbilityState.SourceWeapon == EffectState.ApplyEffectParameters.ItemStateObjectRef)
 	{
-		ShotInfo.ModType = eHit_Success;
-		ShotInfo.Reason = FriendlyName;
-		ShotInfo.Value = Aim_Bonus;
-		ShotModifiers.AddItem(ShotInfo);
-
-		ShotInfo.ModType = eHit_Crit;
-		ShotInfo.Reason = FriendlyName;
-		ShotInfo.Value = Crit_Bonus;
-		ShotModifiers.AddItem(ShotInfo);
+		if(default.WEAPON_CATEGORIES_HIGH_BONUS.Find(SourceWeaponState.GetWeaponCategory()) != INDEX_NONE)
+		{
+			ShotInfo.ModType = eHit_Success;
+			ShotInfo.Reason = FriendlyName;
+			ShotInfo.Value = Aim_Bonus * default.HIGH_BONUS_MODIFIER;
+			ShotModifiers.AddItem(ShotInfo);
+		
+			ShotInfo.ModType = eHit_Crit;
+			ShotInfo.Reason = FriendlyName;
+			ShotInfo.Value = Crit_Bonus* default.HIGH_BONUS_MODIFIER;
```

Comment:
Why not put the aim and crit values in the `if` blocks and the rest after them? There's a lot of code duplication here since the `ModType`s, `Reason`s and `AddItem()`s are the same for all three branches.

Reply:
I did it because Shotinfo is one variable, and it would just override each other. But yeah, I think I'm going to give crit a seperate variable.